src/main/java/org/example/gabartischolarsqbbe/controller/AuthController.java

```diff
@@ -0,0 +1,82 @@
+package org.example.gabartischolarsqbbe.controller;
+
+import org.example.gabartischolarsqbbe.dto.LoginRequest;
+import org.example.gabartischolarsqbbe.dto.SignupRequest;
+import org.example.gabartischolarsqbbe.entity.User;
+import org.example.gabartischolarsqbbe.service.AuthService;
+import org.example.gabartischolarsqbbe.config.JwtUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    @Autowired
+    private AuthService authService;
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> registerUser(@RequestBody SignupRequest signupRequest) {
+        try {
+            User user = authService.registerUser(
+                signupRequest.getUsername(),
+                signupRequest.getEmail(),
+                signupRequest.getPassword()
+            );
+
+            String token = jwtUtil.generateToken(user.getUsername());
+
+            Map<String, Object> response = new HashMap<>();
+            response.put("message", "User registered successfully!");
+            response.put("userId", user.getId());
+            response.put("username", user.getUsername());
+            response.put("email", user.getEmail());
+            response.put("token", token);
+
+            return ResponseEntity.ok(response);
+
+        } catch (RuntimeException e) {
+            Map<String, String> errorResponse = new HashMap<>();
+            errorResponse.put("error", e.getMessage());
+            return ResponseEntity.badRequest().body(errorResponse);
+        }
+    }
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@RequestBody LoginRequest loginRequest) {
```

Comment:
As the previous comment, so maybe rename the endpoint to `\authenticate` or rename the method to `signin`